runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -251,6 +251,48 @@ J9::CodeGenerator::fastpathAcmpHelper(TR::Node *node, TR::TreeTop *tt, const boo
    // next treetop and then at the current one
    TR::Block* prevBlock = tt->getEnclosingBlock();
    TR::Block* targetBlock = prevBlock->splitPostGRA(tt->getNextTreeTop(), cfg, true, NULL);
+
+   // Move call tree to end of block, along with any stores of the result of the call, if they're
+   // not already at the end.  The call has to be split into its own block, but splitPostGRA above
+   // might have left register stores and stores into temporaries that must stay with the previous block
+   TR::TreeTop* prevBlockExit = prevBlock->getExit();
+   TR::TreeTop* afterTT = tt->getNextTreeTop();
+
+   if (afterTT != prevBlockExit)
+      {
+      if (trace)
+         {
+         traceMsg(comp, "Moving treetop containing node n%dn [%p] associated with acmp helper call to end of prevBlock in preparation of final block split\n", tt->getNode()->getGlobalIndex(), tt->getNode());
+         }
+
+      // Remove TreeTop for call node
+      tt->unlink(false);
+      TR::TreeTop* callTreesTail = tt;
+
+      while (afterTT != prevBlockExit)
+         {
+         TR::TreeTop* nextTT = afterTT->getNextTreeTop();
+         TR::ILOpCodes op = afterTT->getNode()->getOpCodeValue();
+
+         if ((op == TR::iRegStore || op == TR::istore) && afterTT->getNode()->getFirstChild() == node)
+            {
+            if (trace)
+               {
+               traceMsg(comp, "Moving treetop containing node n%dn [%p] associated with acmp helper call to end of prevBlock in preparation of final block split\n", afterTT->getNode()->getGlobalIndex(), afterTT->getNode());
```

Comment:
Should this trace be updated to differentiate from the trace message at line L265 to show this one is more specific to store nodes?